client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodParser.java

```diff
@@ -265,9 +285,26 @@ public String path(Object[] methodArguments) {
     public Iterable<HttpHeader> headers(Object[] swaggerMethodArguments) {
         final HttpHeaders result = new HttpHeaders(headers);
 
-        final Iterable<EncodedParameter> substitutedHeaders = getEncodedParameters(headerSubstitutions, swaggerMethodArguments, null);
-        for (final EncodedParameter substitutedHeader : substitutedHeaders) {
-            result.set(substitutedHeader.name(), substitutedHeader.encodedValue());
+        if (headerSubstitutions != null) {
+            for (Substitution headerSubstitution : headerSubstitutions) {
+                final int parameterIndex = headerSubstitution.methodParameterIndex();
+                if (0 <= parameterIndex && parameterIndex < swaggerMethodArguments.length) {
+                    final Object methodArgument = swaggerMethodArguments[headerSubstitution.methodParameterIndex()];
+                    if (methodArgument instanceof Map) {
```

Comment:
If you want, you can get the full static type information from methodParser

Reply:
Strictly speaking, the keys just need to be something that converts gracefully to String.